src/Cake.Core/IO/DirectoryPath.cs

```diff
@@ -151,5 +151,70 @@ public static DirectoryPath FromString(string path)
         {
             return new DirectoryPath(path);
         }
+
+        /// <summary>
+        /// Get the relative path to another directory.
+        /// </summary>
+        /// <param name="to">The target directory path.</param>
+        /// <returns>A <see cref="DirectoryPath"/>.</returns>
+        public DirectoryPath GetRelativePath(DirectoryPath to)
```

Comment:
Could we break the logic of this method to it's own file. Mostly an aesthetic adjustment, but would follow how we've done with the PathCollapser stuff.